src/controllers/friendRequestController.js

```diff
@@ -0,0 +1,159 @@
+        }
+        const requestAlreadyExists = await tableContainsLink("friend_requests", "user1_id", "user2_id", user1_id, user2_id, friend_requests_cache)
+        const usersAlreadyFriends = friendship_cache.has(String(user1_id) + '-' + String(user2_id))
+
+        if (usersAlreadyFriends) {
+            res.status(400).json({ status: 400, message: "Invalid request: Users are already friends" })
+        }
+        else if (requestAlreadyExists) {
+            res.status(400).json({ status: 400, message: "Invalid request: The request object already exists, withdraw it, or wait till it is accepted/denied" })
+        }
+        else {
+
+            client.query("INSERT INTO friend_requests (user1_id, user2_id, request_status) VALUES ($1, $2, $3) RETURNING *",
+                [user1_id, user2_id, "pending"],
+                (err, results) => {
+                    if (!err) {
+                        res.status(201).json(results.rows)
+                    }
+                    else {
+                        console.log(err)
+                        res.status(500).json({
+                            status: 500, message: "unknown error occurred while creating friend request object"
+                        })
+                    }
+                }
+            )
+        }
+    } catch (error) {
+        next(error)
+    }
+}
+
+
+async function deleteFriendRequestObjectById(req, res) {
+    const request = await friendRequestLookup(req.params.friend_request_id)
+    const {user1_id, user2_id} = request;
+    const requestExists = await tableContainsLink("friend_requests", "user1_id", "user2_id", user1_id, user2_id, friend_requests_cache)
+
+    if (requestExists) {
+
+        if (req.user.id == user1_id || req.user.id == user2_id) {
+
+            client.query("DELETE FROM friend_requests WHERE request_id = $1 RETURNING *", [req.params.friend_request_id],
+
+                (err, results) => {
+                    if (!err) {
+                        res.status(200).json({
+                            status: 200, message: "request object deleted successfully"
+                        })
+                        removeElementFromSet(friend_requests_cache, String(results.rows[0].user1_id) + "-" + String(results.rows[0].user2_id))
+                    }
+                    else {
+                        res.status(500).json({
+                            status: 500, message: "unknown error occurred while deleting request object"
+                        })
+                    }
+                }
+            )
+
+        }
+        else {
+            res.status(400)
+                .json({ status: 400, message: "Invalid request: Only the users who are friends related to the request can perform this action" })
+        }
+
+    }
+    else {
+        res.status(400).json({ status: 400, message: "Invalid request: No such community request object" })
+    }
+}
+
+const updateFriendRequestStatus = async (req, res,next) => {
```

Comment:
Make a generic update function
and to alter whether the request is accepted or not, make a different function
You can take care of routing using headers, etc
Refer the communityRequestRouter, controller